figures/generate-figures.sh

```diff
@@ -43,6 +43,39 @@ mkdir -p pdfs/supp/figs6/panels
 Rscript --vanilla scripts/color_palettes.R
 Rscript -e "rmarkdown::render('mapping-histology-labels.Rmd', clean = TRUE, params = list(release = 'release-v22-20220505'))"
 
+### Make sure relevant scratch/ files are up to date ------------------------------
+
+# We will check that necessary input files exist and are <=10 days old,
+#  assuming that <=10 days is likely up-to-date.
+#   If input files are older, exit with code 0 and prompt to run the module.
+MAX_DAYS=10
+
+# Check for an oncoprint file:
+oncoprint_file_check=${scratch_dir}/oncoprint_files/primary_only_maf.tsv # use one of the input files in the check
+oncoprint_scratch_days=$((($(date +%s) - $(date +%s -r "$oncoprint_file_check")) / 86400))  # https://unix.stackexchange.com/questions/102691/get-age-of-given-file
+if [ ! -f $oncoprint_file_check ]; then
+  echo "The 'oncoprint-landscape' module scratch files do not all exist. Please re-run the 'oncoprint-landscape' module first, or run the script 'scripts/run-manuscript-analyses.sh'."
+  exit 0
+fi
+if [ $oncoprint_scratch_days -gt ${MAX_DAYS} ]; then
+  echo "WARNING: The 'oncoprint-landscape' module scratch files may be out of date. You may want to re-run the 'oncoprint-landscape' module first, or run the script 'scripts/run-manuscript-analyses.sh'."
+fi
+
+# Check for snv-callers databases:
+snv_pbta_db_check=${scratch_dir}/snv-callers/snv_db.sqlite
+snv_tcga_db_check=${scratch_dir}/snv-callers/tcga_snv_db.sqlite
+snv_pbta_scratch_days=$((($(date +%s) - $(date +%s -r "$snv_pbta_db_check")) / 86400))  # https://unix.stackexchange.com/questions/102691/get-age-of-given-file
+snv_tcga_scratch_days=$((($(date +%s) - $(date +%s -r "$snv_tcga_db_check")) / 86400))  # https://unix.stackexchange.com/questions/102691/get-age-of-given-file
+if [ ! -f $snv_pbta_db_check ] || [ ! -f $snv_tcga_db_check ]; then
+  echo "The 'snv-callers' module scratch databases do not exist. Please re-run the 'snv-callers' module first, or run the script 'scripts/run-manuscript-analyses.sh'."
+  exit 0
+fi
```

Comment:
This check should probably be before the date calculations, or I expect they will fail. 
Also, since this is an error, `exit 1`

Reply:
done!